ls1intum/Artemis
# `Development`: Add more test servers and several improvements

> [!WARNING]
> Please note: It is intentional that this PR currently blocks all Testservers (except TS9). Please use the old testservers via Bamboo in the meantime.




### Checklist
#### General



### Motivation and Context


1. We want to add new setups to the new testservers. For this we need more testservers
2. We want to have an easy way to view, which testservers are currently available.

### Description

1. I added TS1-3 here. TS4-6 have also been redeployed to have new setups
2. I added badges that show the current status of each testserver. This is also added in every PR description from now on.
3. I also added a docker compose file for LocalCI deployments, as this needs additional config
4. I also fixed some smaller issues with the postgres deployment
5. Added documentation about the servers here: https://confluence.ase.in.tum.de/x/lVGBAQ 

### Steps for Testing

1. Deploy this PR to a test server via labels
6. Deployment succeeds
7. Badge updates (can take a little bit, because GitHub caches it, viewing it directly should always be up to date)

Note: You can't test the label update on PR close and label remove. These actions always use workflows from develop. I tested them in my fork of Artemis and they worked as expected.

#### Exam Mode Testing



N/A

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
Red = Currently locked, Green = Currently available

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






N/A

### Screenshots


N/A
